### PR TITLE
feat: store v3 return pubsub topics

### PR DESCRIPTION
**Pull request by SionoiS**
This PR is meant to unblock Waku Sync work.

Before Store v3 did not return the pubsub topics messages were sent on, now it does.

Follows updated https://github.com/vacp2p/rfc/pull/665/commits/5d69ab58b34d32af99e60b109ad784ad1dd9e345

### CODE REVIEW

File: waku/waku_archive/common.nim [hunk 1]
```diff
@@ -56,6 +56,7 @@ type
   ArchiveResponse* = object
     hashes*: seq[WakuMessageHash]
     messages*: seq[WakuMessage]
+    topics*: seq[PubsubTopic]
```
Comment (jm-clius) [inline waku/waku_archive/common.nim:59]: Mmm. Wouldn't it be better to return seq[(message, topic)] to keep the 1:1 association explicit?

**Review by jm-clius — APPROVED** [review]
Approving, but I do think the comment below should be addressed.

**Review by gabrielmer — APPROVED** [review]
LGTM 🤩

**Comment by SionoiS** [comment]
We got a segfault in CI but i can't reproduce locally. :|

**Comment by SionoiS** [comment]
Will merge even with CI red as it's unrelated and the issue is being investigated.

**Pull request merged**